src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client_test.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+import time
+from typing import List, Tuple
+import unittest
+
+import grpc.experimental
+import xds_interop_client
+import xds_interop_server
+
+from src.proto.grpc.testing import empty_pb2
+from src.proto.grpc.testing import messages_pb2
+from src.proto.grpc.testing import test_pb2
+from src.proto.grpc.testing import test_pb2_grpc
+import src.python.grpcio_tests.tests.unit.framework.common as framework_common
+
+_CLIENT_PATH = os.path.abspath(os.path.realpath(xds_interop_client.__file__))
+_SERVER_PATH = os.path.abspath(os.path.realpath(xds_interop_server.__file__))
+
+_METHODS = (
+    (messages_pb2.ClientConfigureRequest.UNARY_CALL, "UNARY_CALL"),
+    (messages_pb2.ClientConfigureRequest.EMPTY_CALL, "EMPTY_CALL"),
+)
+
+_QPS = 100
+_NUM_CHANNELS = 20
+
+_TEST_ITERATIONS = 10
+_ITERATION_DURATION_SECONDS = 1
+_SUBPROCESS_TIMEOUT_SECONDS = 2
+
+
+@contextlib.contextmanager
+def _start_python_with_args(
+    file: str, args: List[str]
+) -> Tuple[subprocess.Popen, tempfile.TemporaryFile, tempfile.TemporaryFile]:
+    with tempfile.TemporaryFile(mode='r') as stdout:
+        with tempfile.TemporaryFile(mode='r') as stderr:
+            proc = subprocess.Popen((sys.executable, file) + tuple(args),
+                                    stdout=stdout,
+                                    stderr=stderr)
+            yield proc, stdout, stderr
+
+
+def _dump_stream(process_name: str, stream_name: str,
+                 stream: tempfile.TemporaryFile):
+    sys.stderr.write(f"{process_name} {stream_name}:\n")
+    stream.seek(0)
+    sys.stderr.write(stream.read())
+
+
+def _dump_streams(process_name: str, stdout: tempfile.TemporaryFile,
+                  stderr: tempfile.TemporaryFile):
+    _dump_stream(process_name, "stdout", stdout)
+    _dump_stream(process_name, "stderr", stderr)
+    sys.stderr.write(f"End {process_name} output.\n")
+
+
+def _test_client(server_port: int, stats_port: int, qps: int,
```

Comment:
optional: since we are asserting in this function, we could make it part of `XdsInteropClientTest` to access the `unittest` assertions.

Reply:
Done.